draft-ietf-mls-protocol.md

```diff
@@ -5573,12 +5177,12 @@ If two members' leaf nodes have the same signature key, for example, then the
 data origin authentication properties afforded by signatures within the group
 are degraded.
 
-Uniqueness of keys in leaf nodes is assured by explicit checks on leaf nodes
-being added to the tree by Add or Update proposals, or in the `path` field of a
+Uniqueness of keys in leaf nodes is assured by explicitly checking each leaf node
+being added to the tree by in Add proposal, in an Update proposals, or in the `path` field of a
```

Comment:
```suggestion
being added to the tree by in Add proposals, in Update proposals, or in the `path` field of a
```

Reply:
Reworded to:

> ... each leaf node as it is added to the tree, whether in an Add proposal, in an Update proposal, or in the `path` field of a Commit.

That way everything is singular.  Also got rid of "by in".